autowarefoundation/autoware.universe
# feat(behavior_path_planner): revise lane change module

## Description

Previous lane change logic checks the distance between ego to the object using l2Norm. However, some information lost might occurs for example the object is on the left or right etc.

The lane change module is currently revised internally to better handle the dynamic environment. Instead of using l2norm, the distance is separated into lateral distance evaluation and longitudinal distance evaluation. the longitudinal distance evaluation takes into account of collision via predicted path.

This PR aims to realize the newly revised functions.

For configuration, add the following https://github.com/tier4/autoware_launch/pull/355
For visualization, add the following #1140 

## Related links



## Tests performed

1. `rtc_auto_approval` for right and left lane change needs to be disable
2. Run the PSIM.
3. Place ego starting position and goal. Goal must be on the adjacent lane to simulate lane change process.
4. Place an object on the target lane.
![Screenshot from 2022-06-21 13-40-31](https://user-images.githubusercontent.com/93502286/174717952-d6b0a44c-4751-431d-9a2d-ba7df5a9f181.png)
5. If ego is within object vicinity and there is potential of collision via predicted path comparison, no path appears.
![Screenshot from 2022-06-21 13-41-26](https://user-images.githubusercontent.com/93502286/174718138-c5134714-2f32-425c-85bf-fbed9d94f3aa.png)

For configuration, cherry-pick/pull the following https://github.com/tier4/autoware_launch/pull/355

Documentation #1385 

## Notes for reviewers

1. Reviewer should also confirm that the solution will not break the normal avoidance behavior.
2. Please add the new parameters to the config file.
3. Force lane change is removed for now.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
